pkg/promclient/promclient.go

```diff
@@ -172,26 +172,24 @@ func (c *Client) ExternalLabels(ctx context.Context, base *url.URL) (labels.Labe
 
 	body, _, err := c.req2xx(ctx, &u, http.MethodGet, nil)
 	if err != nil {
-		return nil, err
+		return labels.EmptyLabels(), err
 	}
 	var d struct {
 		Data struct {
 			YAML string `json:"yaml"`
 		} `json:"data"`
 	}
 	if err := json.Unmarshal(body, &d); err != nil {
-		return nil, errors.Wrapf(err, "unmarshal response: %v", string(body))
+		return labels.EmptyLabels(), errors.Wrapf(err, "unmarshal response: %v", string(body))
 	}
 	var cfg struct {
 		GlobalConfig config.GlobalConfig `yaml:"global"`
 	}
 	if err := yaml.Unmarshal([]byte(d.Data.YAML), &cfg); err != nil {
-		return nil, errors.Wrapf(err, "parse Prometheus config: %v", d.Data.YAML)
+		return labels.EmptyLabels(), errors.Wrapf(err, "parse Prometheus config: %v", d.Data.YAML)
 	}
 
-	lset := cfg.GlobalConfig.ExternalLabels
-	sort.Sort(lset)
```

Comment:
No sorting here anymore? I am not sure if it is expected or we know external labels in the response will always be sorted?

Reply:
Oh for stringlabels we dont need that but if we are not building with it we probably need it still! Thanks